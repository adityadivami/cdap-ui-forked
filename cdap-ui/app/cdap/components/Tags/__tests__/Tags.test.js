import React from 'react';
import { cleanup, render, fireEvent, wait, waitForElement } from 'react-testing-library';
import Tags from 'components/Tags';
import Mousetrap from 'mousetrap';
import 'jest-dom/extend-expect';

afterEach(cleanup);

jest.mock('api/metadata');

const testEntity = {
  type: 'application',
  id: 'test'
};

describe('Tags', () => {
  it('renders correctly', () => {
    const { container } = render(<Tags entity={testEntity} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('adds new tag when user inputs a tag and presses Enter', async () => {
    const TEST_TAG = 'UI_TEST_TAG';
    const { getByTestId, queryByTestId, getByText, container } = render(<Tags entity={testEntity} />);
    await wait(() => !queryByTestId('loading-icon')); // wait until loading icon is gone, which means async call is done

    const plusButtonElem = getByTestId('tag-plus-button');
    fireEvent.click(plusButtonElem, {container}); // click on plus button to show input box
    await waitForElement(() => getByTestId('tag-input'));

    const inputElem = getByTestId('tag-input');
    expect(inputElem).toHaveTextContent('');
    fireEvent.change(inputElem, {target: {value: TEST_TAG}});
    Mousetrap.trigger('return'); // enter value, and then press Enter to add tag
    await waitForElement(() => getByText(TEST_TAG));

    expect(inputElem).not.toBeInTheDocument(); // input elem is hidden after user presses Enter
    expect(getByText(TEST_TAG)).toBeVisible();
  });
});

