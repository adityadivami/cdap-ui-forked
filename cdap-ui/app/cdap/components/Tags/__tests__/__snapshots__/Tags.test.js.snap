// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div
  class="tags-holder"
>
  <strong>
    features.Tags.labelWithCount:
  </strong>
  <i>
    features.Tags.notags
  </i>
  <span
    class="tags-list"
  >
    <span />
    <span />
  </span>
  <span
    class="btn btn-primary plus-button-container"
    data-testid="tag-plus-button"
  >
    <svg
      class="icon-svg icon-plus text-white"
    >
      <use
        xlink:href="http://localhost/#symbol-defs_icon-plus"
      />
      <use
        xlink:href="http://localhost/#symbol-defs_new_icon-plus"
      />
    </svg>
  </span>
</div>
`;
